aws/table_aws_backup_framework.go

```diff
@@ -0,0 +1,240 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/backup"
+
+	"github.com/turbot/go-kit/types"
+	"github.com/turbot/steampipe-plugin-sdk/v3/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin/transform"
+)
+
+//// TABLE DEFINITION
+func tableAwsBackupFramework(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_backup_framework",
+		Description: "AWS Backup Framework",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("framework_name"),
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: isNotFoundError([]string{"InvalidParameterValueException"}),
+			},
+			Hydrate: getAwsBackupFramework,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listAwsBackupFrameworks,
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "framework_name",
+				Description: "The unique name of a backup framework.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "arn",
+				Description: "An Amazon Resource Name (ARN) that uniquely identifies a backup framework resource.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("FrameworkArn"),
+			},
+			{
+				Name:        "framework_description",
+				Description: "An optional description of the backup framework.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "deployment_status",
+				Description: "The deployment status of a backup framework.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "creation_time",
+				Description: "The date and time that a framework was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "number_of_controls",
+				Description: "The number of controls contained by the framework.",
+				Type:        proto.ColumnType_INT,
+				Hydrate:     getNumberOfControls,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "framework_status",
+				Description: "A framework consists of one or more controls. Each control governs a resource, such as backup plans, backup selections, backup vaults, or recovery points.",
```

Comment:
This description doesn't describe the column well. From the API description, we could probably take some of its content and turn it into:
```
The framework status based on recording statuses for resources governed by the framework (ACTIVE | PARTIALLY_ACTIVE | INACTIVE | UNAVAILABLE).
```